x/msgfees/types/fee.go

```diff
@@ -9,11 +9,19 @@ const (
 	NhashDenom string = "nhash"
 )
 
-// SplitAmount returns split of Amount to be used for coin recipient and one for payout of fee, NOTE: this should only be used if a Recipient address exists
-func SplitAmount(coin sdk.Coin) (recipientCoin sdk.Coin, feePayoutCoin sdk.Coin) {
-	addFeeToPay := coin.Amount.Uint64()
-	addFeeToPay /= 2
-	feePayoutCoin = sdk.NewCoin(coin.Denom, sdk.NewIntFromUint64(addFeeToPay))
-	recipientCoin = coin.Sub(feePayoutCoin)
-	return recipientCoin, feePayoutCoin
+// SplitCoinByBips returns split to recipient and fee module based on basis points for recipient
+func SplitCoinByBips(coin sdk.Coin, bips uint32) (recipientCoin sdk.Coin, feePayoutCoin sdk.Coin, err error) {
+	if bips > 10_000 {
```

Comment:
i must be missing something, bips will default to 0(validate will pass), where are you setting it to `DefaultMsgFeeBips` when recipient is set but bips is not set by the creator?